kahuna/public/js/components/gr-image-metadata/gr-image-metadata.js

```diff
@@ -13,153 +13,150 @@ export const module = angular.module('gr.imageMetadata', [
-        });
-
-        ctrl.updateMetadataField = function (field, value) {
-            return editsService.updateMetadataField(ctrl.image, field, value)
-                .then((updatedImage) => {
-                    if (updatedImage) {
-                        ctrl.image = updatedImage;
-                        updateAbilities(updatedImage);
-                        $rootScope.$emit(
-                          'track:event',
-                          'Metadata',
-                          'Edit',
-                          'Success',
-                          null,
-                          {
-                            field: field,
-                            value: value
-                          }
-                        );
-                    }
-                })
-                .catch(() => {
-                  $rootScope.$emit(
-                    'track:event',
-                    'Metadata',
-                    'Edit',
-                    'Failure',
-                    null,
-                    {
-                      field: field,
-                      value: value
-                    }
-                  );
-
-                    /*
-                     Save failed.
-
-                     Per the angular-xeditable docs, returning a string indicates an error and will
-                     not update the local model, nor will the form close (so the edit is not lost).
-                     Instead, a message is shown and the field keeps focus for user to edit again.
-
-                     http://vitalets.github.io/angular-xeditable/#onbeforesave
-                     */
-                    return 'failed to save (press esc to cancel)';
-                });
-        };
-
-        ctrl.removeImageFromCollection = (collection) => {
-            ctrl.removingCollection = collection;
-            collections.removeImageFromCollection(collection, ctrl.image)
-                .then(() => ctrl.removingCollection = false);
-        };
-
-        ctrl.displayLeases = () => {
-            return ctrl.userCanEdit || ctrl.image.leases > 0;
-        };
-
-        $scope.$on('$destroy', function() {
-            freeUpdateListener();
-        });
+    function isUsefulMetadata(metadataKey) {
+      return ignoredMetadata.indexOf(metadataKey) === -1;
+    }
+
+    function updateAbilities(image) {
+      editsService.canUserEdit(image).then(editable => {
+        ctrl.userCanEdit = editable;
+      });
     }
+
+    const freeUpdateHandler = (updatedImage) => {
+      console.log('image update recvd in gr-image-metadata', updatedImage);
+      ctrl.image = updatedImage;
+      ctrl.usageRights = imageService(ctrl.image).usageRights;
+      ctrl.metadata = updatedImage.data.metadata;
+      ctrl.setUsageCategory(ctrl.usageCategories, ctrl.usageRights.data.category);
+    };
+
+    const freeUpdateListener = $rootScope.$on('images-updated', (e, updatedImages) => updatedImages.map(updatedImage => freeUpdateHandler(updatedImage)));
```

Comment:
This is a little odd – are we only ever expecting a single image in `updatedImages`, or do we just care about the last one? (It seems that `freeUpdateHandler` only cares about the image it's currently receiving). 

If we just care about the last one, can we make that explicit by calling it once with the last element in `updatedImages`?

(v. possible I have wrong end of stick!)

Reply:
I think you're right, but I didn't really want to alter the logic.